backend/portal/serializers.py

```diff
@@ -10,81 +11,78 @@ class Meta:
         fields = "__all__"
 
 
-class PollSerializer(serializers.ModelSerializer):
+class ContentSerializer(serializers.ModelSerializer):
     class Meta:
-        model = Poll
         fields = (
             "id",
             "club_code",
-            "question",
             "created_date",
             "start_date",
             "expire_date",
-            "multiselect",
             "club_comment",
             "admin_comment",
             "status",
             "target_populations",
         )
         read_only_fields = ("id", "created_date")
+        abstract = True
+
+    def _auto_add_target_population(self, validated_data):
+        # auto add all target populations of a kind if not specified
+        if target_populations := validated_data.get("target_populations"):
```

Comment:
May be more efficient to do this using set difference instead of nested loops

Reply:
Good thought, although it doesn't matter here since if you look Kind has like 4 values. Probably the overhead from sets is more costly anyway